doc/setup-commands.rst

```diff
@@ -924,18 +916,30 @@ Miscellaneous options
         # specify this.)
         $ cabal install --constraint="bar source"
 
+    Further, we can specify flag assignments with ``+FLAG`` and ``-FLAG``
+    or enable test (``test``) and benchmark (``bench``) suites:
+
+    ::
+
+        # Require bar to be installed with the foo flag turned on and
+        # the baz flag turned off.
+        $ cabal install --constraint="bar +foo -baz"
+
         # Require that bar have test suites and benchmarks enabled.
         $ cabal install --constraint="bar test" --constraint="bar bench"
 
+    Observe: To specify multiple constraints, you may pass the
+    ``constraint`` option multiple times.
+
     By default, constraints only apply to build dependencies
     (:pkg-field:`build-depends`), build dependencies of build
     dependencies, and so on. Constraints normally do not apply to
     dependencies of the ``Setup.hs`` script of any package
     (:pkg-field:`custom-setup:setup-depends`) nor do they apply to build tools
     (:pkg-field:`build-tool-depends`) or the dependencies of build
     tools. To explicitly apply a constraint to a setup or build
-    tool dependency, you can add a qualifier to the constraint as
-    follows:
+    tool dependency, you can add a qualifier ``setup`` or ``any``
```

Comment:
I really dislike this style when, in this case, the definitions of `any` and `setup` are given in the comments of an example. There's description in the text above but it doesn't name the options!

Reply:
Mmh, I don't like syntax _just_ be introduced by examples.  I like the style _overview followed by details_.  Graphically, I want the syntax to stick out, and in the example, everything is highlighted the same, so the emphasis on the important new syntax isn't there.

Reply:
There _is_ highlighting of `any` and `setup` in the example due to putting them in single quotes and using the `console` highlighter.  However, this highlighter is broken in that it puts __strong__ emphasis on irrelevant words like _if_ and _in_ in the highlighted comment.  I find this very irritating if my attention is strongly drawn to purely administrative words in a sentence.  
A bigger rewrite of this section would move out the text from the console comments into the main body, fixing both your and my concern.  However, I don't want to go so far now.

Reply:
I just think that it's a bit of a pity to have a very nice and short explanation in comments as opposed to the main text. But I don't have any concrete suggestion here, so perhaps it's all right.